curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md

```diff
@@ -461,15 +461,23 @@ cashInput.value = `${randomCash / 100}`;
 let changeLeft = randomChange;
 const _expectedChangeDue = [];
 const _cashInDrawer = [];
-for (const [denominationName, denomination] of _money) {
+for (let i = 0; i < _money.length; i++) {
+  const [denominationName, denomination] = _money[i];
+
   const maxCountInChange = Math.floor(changeLeft / denomination);
-  // If denomination can complete required changeLeft, available amount in drawer cannot
-  // equal the maximum. Otherwise count in drawer can be greater than maximum count in change.
-  const drawerCount = _randomNumber(
-    changeLeft % denomination === 0 ? Math.min(15, maxCountInChange - 1) : 15
-  );
+
+  let drawerCount;
+  if (i === 0) {
+    drawerCount = _randomNumber(Math.min(15, maxCountInChange));
+  } else {
+    const previousDrawerTotal = _cashInDrawer[_cashInDrawer.length - 1][1] * 100;
+    const maxDrawerCount = Math.floor(previousDrawerTotal / denomination);
+    drawerCount = _randomNumber(Math.min(15, Math.min(maxCountInChange, maxDrawerCount - 1)));
```

Comment:
If I'm reading it correctly, `drawerCount` will always be `0` after amount of the previous denomination was `0`.

Reply:
Yes you are right, i need to think more about this i guess, for now i close the pr